extensions/blocks/dialogue/dialogue.php

```diff
@@ -0,0 +1,158 @@
+<?php
+/**
+ * Dialogue Block.
+ *
+ * @since 9.3.0
+ *
+ * @package Jetpack
+ */
+
+namespace Automattic\Jetpack\Extensions\Dialogue;
+
+use Automattic\Jetpack\Blocks;
+use Jetpack_Gutenberg;
+
+const FEATURE_NAME = 'dialogue';
+const BLOCK_NAME   = 'jetpack/' . FEATURE_NAME;
+
+/**
+ * Registers the block for use in Gutenberg
+ * This is done via an action so that we can disable
+ * registration if we need to.
+ */
+function register_block() {
+	$deprecated = function_exists( 'gutenberg_get_post_from_context' );
+	$uses       = $deprecated ? 'context' : 'uses_context';
+	Blocks::jetpack_register_block(
+		BLOCK_NAME,
+		array(
+			'render_callback' => __NAMESPACE__ . '\render_block',
+			$uses             => array(
+				'jetpack/conversation-participants',
+				'jetpack/conversation-showtimestamp',
+			),
+		)
+	);
+}
+add_action( 'init', __NAMESPACE__ . '\register_block' );
+
+/**
+ * Dialogue block registration/dependency declaration.
+ *
+ * @param array  $attrs    Array containing the Dialogue block attributes.
+ * @param string $block_content String containing the Dialogue block content.
+ * @param object $block Block object data.
+ *
+ * @return string
+ */
+function render_block( $attrs, $block_content, $block ) {
+	Jetpack_Gutenberg::load_assets_as_required( FEATURE_NAME );
+
+	$default_participants = json_decode(
+		// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+		file_get_contents( JETPACK__PLUGIN_DIR . 'extensions/blocks/conversation/participants.json' ),
+		true
+	);
```

Comment:
Since it's a relatively small list, couldn't we build that array of default participants right here in PHP instead of relying on an external file? That would allow us to translate those default values.

This is what we did for the Business Hours block, for example:
https://github.com/Automattic/jetpack/blob/ebec0a8fb35ffac4766cce36646ebf01c5d56063/extensions%2Fblocks%2Fbusiness-hours%2Fbusiness-hours.php#L31-L92

Reply:
Well... the idea behind creating a file for these data is to be usable from the client-side too, or order to guarantee the parity between both sides. We'll have to figure out how to deal with translation, though. Could we consider handling it in a follow-up PR?

Reply:
Yeah, that can be handled in a follow-up PR. I think you could have something that can be used on the client side and here by creating a new v2 endpoint that would return the data you need. This way it would all be translated.